File: services/nodejs-microservices/payment-service/src/infrastructure/web/dto.ts
import { Payment, Refund } from '../../domain/models/Payment';

export interface ProcessPaymentRequest {
  orderId: string;
  userId: string;
  amount: number;
  method: string;
  currency?: string;
}

export interface RefundRequest {
  reason?: string;
  amount?: number;
}

export function toPaymentResponse(payment: Payment) {
  return {
    id: payment.id,
    orderId: payment.orderId,
    userId: payment.userId,
    amount: payment.amount,
    currency: payment.currency,
    method: payment.method,
    status: payment.status,
    providerTransactionId: payment.providerTransactionId,
    failureReason: payment.failureReason,
    createdAt: payment.createdAt.toISOString(),
    updatedAt: payment.updatedAt.toISOString(),
  };
}

export function toRefundResponse(refund: Refund) {
  return {
    id: refund.id,
    paymentId: refund.paymentId,
    amount: refund.amount,
    reason: refund.reason,
    status: refund.status,
    createdAt: refund.createdAt.toISOString(),
  };
}

File: services/python-microservices/payment-service/app/infrastructure/web/dto.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from decimal import Decimal


class CreatePaymentRequest(BaseModel):
    orderId: str
    userId: str
    amount: Decimal
    method: str = "CREDIT_CARD"
    currency: str = "USD"


class RefundRequest(BaseModel):
    amount: Optional[Decimal] = None
    reason: Optional[str] = None


class PaymentResponse(BaseModel):
    id: str
    order_id: str
    user_id: str
    amount: Decimal
    currency: str
    method: str
    status: str
    transaction_id: Optional[str] = None
    failure_reason: Optional[str] = None
    retry_count: int
    created_at: datetime
    updated_at: datetime


class RefundResponse(BaseModel):
    id: str
    payment_id: str
    amount: Decimal
    reason: Optional[str] = None
    status: str
    created_at: datetime
